### DISCUSSION TITLE
[Feature Request] Doc on how to run docker inside a microVM

### DISCUSSION THREAD

**Discussion by dianpopa** [firecracker-microvm/firecracker#4038] [General]
# Feature Request

Firecracker cannot run a docker instance inside a microVM with the guest kernel we are currently offering. It would be nice to offer some documentation around the guest setup is needed in order to achieve that.

## Describe the desired solution

* documentation around the guest configuration options needed to run docker inside a microVM
* an docker enabled guest kernel example

## Additional context

Can be found here: https://github.com/firecracker-microvm/firecracker/issues/3476.

## Checks

- [x] Have you searched the Firecracker Issues database for similar requests?
- [x] Have you read all the existing relevant Firecracker documentation?
- [x] Have you read and understood Firecracker's core tenets?

**Comment by JonathanWoollett-Light** [comment]
Running docker inside a guest relates to the KVM and how a user configures Firecracker. We generally only provide minimal examples covering the setup, our goal is to keep our documentation up-to-date and simple.

To offer an example here would require us to store additional artefacts and apply maintenance effort. To avoid this and the risk of it becoming out of date we think this issue makes sense as a discussion as community contribution could be useful here to provide examples people have tried.

**Comment by kalyazin** [comment]
As well summarised by @JonathanWoollett-Light we do not plan to include official documentation on docker setup in a microVM, however we welcome instructions from the community to be posted in this discussion that other users may benefit from.
